pipeline/config_test.go

```diff
@@ -0,0 +1,263 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/noop"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
+	"github.com/open-telemetry/opentelemetry-log-collection/testutil"
+)
+
+func newDummyOpConfig(dummyID string, dummyType string) *operator.Config {
+	return &operator.Config{
+		Builder: &noop.NoopOperatorConfig{
+			TransformerConfig: helper.NewTransformerConfig(dummyID, dummyType),
+		},
+	}
+}
+
+type deduplicateTestCase struct {
+	name        string
+	ops         func() Config
+	expectedOps Config
+}
+
+func TestDeduplicateIDs(t *testing.T) {
```

Comment:
Would it be reasonable to write another set of tests that validate how this affects plugins?

Reply:
I have a set of tests within the `config_test` of plugin tests but I did just update them to remove the parts that were hardcoded in.